Wikipedia/Code/WebViewController.m

```diff
@@ -223,7 +223,13 @@ - (void)handleLinkClickedScriptMessage:(NSDictionary *)messageDict {
                                                    url = [NSURL wmf_URLWithSiteURL:self.article.url escapedDenormalizedInternalLink:href];
                                                }
                                                url = [url wmf_urlByPrependingSchemeIfSchemeless];
-                                               [(self).delegate webViewController:(self) didTapOnLinkForArticleURL:url];
+                                               [self.article.dataStore.articleSummaryController.fetcher fetchSummaryForArticleWithKey:[url wmf_databaseKey] priority:NSURLSessionTaskPriorityHigh completion:^(WMFArticleSummary * _Nullable summary, NSURLResponse * _Nullable response, NSError * _Nullable error) {
```

Comment:
I'm concerned about introducing this delay without also addressing https://phabricator.wikimedia.org/T225545 - could this be handled the same way we currently handle a `WMFFetcher.unexpectedResponseError` and show an external url (line 1319). The namespace could be checked there - might require more info in the callback from the fetcher

Reply:
Alternatively, this could be handled separately either with T225545 or in a separate PR, the fix for `stringByRemovingPercentEncoding` looks good

Reply:
Something like this https://github.com/wikimedia/wikipedia-ios/pull/3196/commits/8b19351722c3021e671e5ac0d5f2c97d53f441bc?

Reply:
Yup! Looks good